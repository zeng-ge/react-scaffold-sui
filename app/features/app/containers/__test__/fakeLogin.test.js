import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import ConnectFakeLogin, { FakeLogin } from '../FakeLogin'

describe('FakeLogin', () => {

  test('should FakeLogin component show login <a>己登录, 去首页</a>', () => {
    const user = { userId: '123' }
    const component = shallow(<FakeLogin user={user} />)
    expect(component.find('a').first().text()).toBe('己登录, 去首页')
  })

  test('should FakeLogin component show login <a>己登录, 去首页</a>', () => {
    const component = shallow(<FakeLogin />)
    expect(component.find('span').first().text()).toBe('未登录')
  })

  test('should FakeLogin logined', () => {
    const store = global.getStore({
      app: {
        user: {
          userId: '123',
        },
      },
    })
    expect(renderer.create(<ConnectFakeLogin store={store} />).toJSON()).toMatchSnapshot()
  })

  test('should FakeLogin not login', () => {
    const store = global.getStore({ app: {}})
    expect(renderer.create(<ConnectFakeLogin store={store} />).toJSON()).toMatchSnapshot()
  })

})
