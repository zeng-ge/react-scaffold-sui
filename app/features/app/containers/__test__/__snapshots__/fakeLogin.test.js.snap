// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FakeLogin should FakeLogin logined 1`] = `
<div
  className="viewport"
>
  <div>
    <div>
      <a
        href="/home"
      >
        己登录, 去首页
      </a>
    </div>
    <a
      href="http://lphomepad-dev.paas.cmbuat.com"
      target="_blank"
    >
      请点击此处模拟登录
    </a>
  </div>
</div>
`;

exports[`FakeLogin should FakeLogin not login 1`] = `
<div
  className="viewport"
>
  <div>
    <div>
      <span>
        未登录
      </span>
    </div>
    <a
      href="http://lphomepad-dev.paas.cmbuat.com"
      target="_blank"
    >
      请点击此处模拟登录
    </a>
  </div>
</div>
`;
